packages/store/addon/-private/system/model/errors.js

```diff
@@ -225,7 +228,10 @@ export default ArrayProxy.extend(Evented, {
     this._add(attribute, messages);
 
     if (wasEmpty && !get(this, 'isEmpty')) {
-      this.trigger('becameInvalid');
+      this._registeredHandlers && this._registeredHandlers.becameInvalid();
+      if (DEBUG && this._has('becameInvalid')) {
```

Comment:
Why do we only trigger this in debug mode?

Reply:
I believe there was a reason around it wouldn't trigger in prod or triggers via a different mechanism in prod, will double check